src/main/java/org/mockbukkit/mockbukkit/inventory/InventoryViewMock.java

```diff
@@ -123,11 +122,10 @@ public void setType(InventoryType type)
 	@Override
 	public void setTitle(@NotNull String title)
 	{
-		if (!this.titleChanged)
-		{
-			this.originalTitle = this.name;
-			this.titleChanged = true;
-		}
+		Preconditions.checkArgument(title != null, "Title cannot be null");
+		// TODO: Paper also implement this 2 validations in set title. This needs to be worked on
+		// Preconditions.checkArgument(this.getPlayer() instanceof Player, "NPCs are not currently supported for this function");
+		// Preconditions.checkArgument(this.getTopInventory().getType().isCreatable(), "Only creatable inventories can have their title changed");
```

Comment:
Why isn't this implemented and only commented?

Reply:
To be honest, i can't remember what was the reason... But trying to uncomment it out now it seems to be resolved by just update 1 unit test. So i've implemented it :)